### DISCUSSION TITLE
Displaying the user's choice on each trial

### DISCUSSION THREAD

**Discussion by 20-alicial** [jspsych/jsPsych#3079] [Q&A] [answered]
Hello! I am new to jsPsych, so forgive me if this is a dumb question. 

I'm using the _image-keyboard-response plugin_ for an image categorization task, and I'm curious if there is a way to display the user's choice on the screen immediately after they make that choice. Would there be a way to display "You have chosen: X" after they make the categorization? Or maybe circle/bold/highlight the choice they made in the prompt parameter?

I know that _categorize-image_ displays feedback on the correctness of the response, so I was thinking of just setting incorrect_text and correct_text to the same message displaying the choice that has been made. But then I'm not sure how to access the user's response and use it in incorrect_text and correct_text.

Thank you!

**Comment by ChristopheBossens** [comment, marked as answer]
You will have to work with [dynamic parameters](https://www.jspsych.org/7.3/overview/dynamic-parameters/#storing-changing-variables-in-the-data) to get access the response. Not sure if your suggested approach will work with the categorize-image plugin, as the parameters are as far as I know evaluated at the start of the trial so you will not be able to access the response and use it within the same trial.

One possible solution I am thinking of is having a second _image-keyboard-response_ trial following the one in which the participant responds, and use dynamic parameters there to display the message.

I did something similar recently in an experiment that showed several images and clicking the image drew a border around the image.  This used the _html-button-response_ plugin. I think it can be adapted to the _image-button-response_ if necessary.

**Reply by 20-alicial** [reply]
Ah, I see. I will try to use the method you suggested with dynamic parameters and a second _image-keyboard-response!_ Thank you :)

**Reply by 20-alicial** [reply]
Update: it worked!
